sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/ApproximateUnique.java

```diff
@@ -394,35 +395,14 @@ public LargestUnique mergeAccumulators(Iterable<LargestUnique> heaps) {
       Iterator<LargestUnique> iterator = heaps.iterator();
       LargestUnique heap = iterator.next();
       while (iterator.hasNext()) {
-        List<Long> largestHashes = iterator.next().extractOrderedList();
-        for (long hash : largestHashes) {
-          if (!heap.add(hash)) {
-            break; // The remainder of this list is all smaller.
-          }
-        }
+        iterator.next().heap.forEach(h -> heap.add(h));
       }
       return heap;
     }
 
     @Override
     public Long extractOutput(LargestUnique heap) {
-      List<Long> largestHashes = heap.extractOrderedList();
```

Comment:
Moved implementation of extractOutput to LargestUnique. There is no reason to assume sampleSize is same for both. As such is based entirely on state within LargestUnique.